### ISSUE TITLE
edgeDB genserver terminates after startup due to MatchError

### ISSUE THREAD

**Issue by artimath** [edgedb/edgedb-elixir#127]
I'm getting an issue when I run EdgeDB.start_link( ). The edgeDB genserver crashes.

Here is a dummy repo where I get the error: https://github.com/artimath/test_edgedb

`edgedb project init
mix deps.get
mix compile
iex -S mix
~~~~
Interactive Elixir (1.14.5) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> {:ok, pid} = EdgeDB.start_link()
{:ok, #PID<0.1990.0>}
iex(2)>
17:18:12.903 [error] GenServer #PID<0.1990.0> terminating
** (MatchError) no match of right hand side value: {:busy, #Reference<0.2001001606.795738119.140031>, %EdgeDB.Pool.Codel{target: 50, interval: 1000, delay: 0, slow: false, next: -576460742214, poll: #Reference<0.2001001606.795607047.140047>, idle_interval: 1000, idle: #Reference<0.2001001606.795607047.140037>}, {-576460742213095083, 0}}
    (db_connection 2.5.0) lib/db_connection/connection_pool.ex:189: DBConnection.ConnectionPool.handle_info/2
    (edgedb 0.5.1) lib/edgedb/pool.ex:219: EdgeDB.Pool.handle_info/2
    (stdlib 4.3.1) gen_server.erl:1123: :gen_server.try_dispatch/4
    (stdlib 4.3.1) gen_server.erl:1200: :gen_server.handle_msg/6
    (stdlib 4.3.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Last message: {:timeout, #Reference<0.2001001606.795607047.140037>, -576460742213094958}
State: %EdgeDB.Pool.State{type: :busy, queue: #Reference<0.2001001606.795738119.140031>, codel: %EdgeDB.Pool.Codel{target: 50, interval: 1000, delay: 0, slow: false, next: -576460742214, poll: #Reference<0.2001001606.795607047.140047>, idle_interval: 1000, idle: #Reference<0.2001001606.795607047.140037>}, ts: {-576460742213095083, 0}, current_concurrency: 0, suggested_concurrency: nil, max_concurrency: nil, conn_sup: #PID<0.1991.0>, conn_mod: EdgeDB.Connection, conn_opts: [owner: #PID<0.1990.0>, queue: #Reference<0.2001001606.795738119.140031>, conn: [mod: EdgeDB.Connection, opts: [backoff_type: :stop, pool: EdgeDB.Pool, timeout: 15000, tls_security: :no_host_verification, address: {"localhost", 10702}, instance_name: "test_edgedb", user: "edgedb", tls_ca: "-----BEGIN CERTIFICATE-----\nMIIC0zCCAbugAwIBAgIRAJ3KY0CWSUuSn3mt6HyRgA8wDQYJKoZIhvcNAQELBQAw\nGDEWMBQGA1UEAwwNRWRnZURCIFNlcnZlcjAeFw0yMzA1MjcxNjU0NTZaFw00MjA3\nMjcxNjU0NTZaMBgxFjAUBgNVBAMMDUVkZ2VEQiBTZXJ2ZXIwggEiMA0GCSqGSIb3\nDQEBAQUAA4IBDwAwggEKAoIBAQCWCUVYuuQD0I8aMe7uuIvec0Wxd8YtoV+FNrAB\nq5xtqRBuhLC15xuJhe7CQxzFvVTt482ED5UnSnBFpa2bvXWqGBaVOBsj/Hj3/GP1\nrhBwmUVbmuld8p2zDHospT/LGa2y5dz4ZyPT2wEN9avJSt/Q0zagOUl70k/LEfAk\nVHvLk3KDNkIOMwHckMVrqj4nojDt29uzxHaaC+p3BHCvcB/pCm53czzRtjqaGuW4\nhdqhjFxR4NLcnAhdLInq3hsU94T31HvlvkK+l3NE4GZHnhlwJ/RObQ3Ri+Cpjhdm\n5MPJhiRHTXOyDnuCscs8A6bAlv6rU9rWwrsHrWSx2HoORxklAgMBAAGjGDAWMBQG\nA1UdEQQNMAuCCWxvY2FsaG9zdDANBgkqhkiG9w0BAQsFAAOCAQEAUNljXnaWaHaI\n5Pg0NzwiZ+x2CHnA530C7AqvbdV7N5IEf316IQw19YyPpfeQtojmmk2ELX7koSor\nLOTlhWcmPmQ5kg/2SLlVzAkjrYFiRf3wlLw1AMSqgZnAwNB+bEquzju/H2HxHS8Y\n4YQbENJ1da0OzN+1D/QQrZafi/2FTZ3/+1TwC/ZkkR7NG+fXC9/tm9WeuGBYL31K\nA5voDU6so/kUKcOEzIcfvjFglxVSCrZq7d2hNlQGayWKj785GL+5zOTPWctN5qgs\nqNumRJvLvLSsD78bezf0yNKGQpspsqsJBJuRZVgdYYmyySpaiiqm1/STDcLNM+4Z\n4Me0Xekr+w==\n-----END CERTIFICATE-----\n", port: 10702, password: "Sp3r5gs02BP0vRyGeZKhoUOP", database: "edgedb", server_settings: %{}]]]}
** (EXIT from #PID<0.1988.0>) shell process exited with reason: an exception was raised:
    ** (MatchError) no match of right hand side value: {:busy, #Reference<0.2001001606.795738119.140031>, %EdgeDB.Pool.Codel{target: 50, interval: 1000, delay: 0, slow: false, next: -576460742214, poll: #Reference<0.2001001606.795607047.140047>, idle_interval: 1000, idle: #Reference<0.2001001606.795607047.140037>}, {-576460742213095083, 0}}
        (db_connection 2.5.0) lib/db_connection/connection_pool.ex:189: DBConnection.ConnectionPool.handle_info/2
        (edgedb 0.5.1) lib/edgedb/pool.ex:219: EdgeDB.Pool.handle_info/2
        (stdlib 4.3.1) gen_server.erl:1123: :gen_server.try_dispatch/4
        (stdlib 4.3.1) gen_server.erl:1200: :gen_server.handle_msg/6
        (stdlib 4.3.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3`

**Comment by nsidnev** [comment]
I suspect this is due to internal changes in the `db_connection` pool. The pool itself is undocumented, so the `EdgeDB.Pool` module implementation may crash depending on the version of `db_connection`. I checked it on version `2.4.2` and everything works there.

Unfortunately, I also found that the original `DBConnection.ConnectionPool` also doesn't work, so I can't suggest you use it instead of the `EdgeDB.Pool`.

For now, I suggest locking the version by adding `{:db_connection, "2.4.2"}` as a dependency. 

I hope to have some time in June to work on this project, fix this issue, and add new features for EdgeDB 3.0, including EdgeDB cloud support, which most, if not all, of the official EdgeDB clients, already have.
